File: CS/WatermarkDemo/WatermarkDemo/Form1.cs
using DevExpress.Utils.Svg;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.ImageEditor;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace WatermarkDemo {
    public partial class Form1 : XtraForm {
        public Form1() {
            InitializeComponent();
            pictureEdit1.ImageEditorDialogShowing += PictureEdit1_ImageEditorDialogShowing;
        }

        void PictureEdit1_ImageEditorDialogShowing(object sender, ImageEditorDialogShowingEventArgs e) {
            e.Commands.Insert(0, new WatermarkCommand() { Image = svgImageCollection1[0] });
            e.Commands.Insert(1, new WatermarkPreset() { Image = svgImageCollection1[1] });
        }

        void btnShowImageEditor_Click(object sender, EventArgs e) {
            pictureEdit1.ShowImageEditorDialog();
        }
    }

    public class WatermarkCommand : IGraphicCommand {
        public virtual SvgImage Image {
            get;
            set;
        }
        public virtual string ToolTip {
            get { return "Add Watermark"; }
        }
        public virtual void Execute(ImageEditorControl editorControl) {
            editorControl.SetActiveTool(new WatermarkToolControl());
        }
    }

    public class WatermarkPreset : WatermarkCommand {
        public override string ToolTip {
            get { return "Watermark Preset"; }
        }
        public override void Execute(ImageEditorControl editorControl) {
            // Perform the WatermarkGraphicOperation operation in code, using the EditController. Users can undo this operation while the Image Editor is active.
            editorControl.EditController.DoOperation(new WatermarkGraphicOperation("devexpress.com", Color.LightBlue, 15, true));
        }
    }
}

File: VB/WatermarkDemo/WatermarkDemo/Form1.vb
Imports DevExpress.Utils.Svg
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.ImageEditor
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace WatermarkDemo
    Partial Public Class Form1
        Inherits XtraForm

        Public Sub New()
            InitializeComponent()
            AddHandler pictureEdit1.ImageEditorDialogShowing, AddressOf PictureEdit1_ImageEditorDialogShowing
        End Sub

        Private Sub PictureEdit1_ImageEditorDialogShowing(ByVal sender As Object, ByVal e As ImageEditorDialogShowingEventArgs)
            e.Commands.Insert(0, New WatermarkCommand() With {.Image = svgImageCollection1(0)})
            e.Commands.Insert(1, New WatermarkPreset() With {.Image = svgImageCollection1(1)})
        End Sub

        Private Sub btnShowImageEditor_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnShowImageEditor.Click
            pictureEdit1.ShowImageEditorDialog()
        End Sub
    End Class
    Public Class WatermarkCommand
        Implements IGraphicCommand
        Public Overridable Property Image As SvgImage Implements IGraphicCommand.Image
        Public Overridable ReadOnly Property ToolTip() As String Implements IGraphicCommand.ToolTip
            Get
                Return "Add Watermark"
            End Get
        End Property
        Public Overridable Sub Execute(ByVal editorControl As ImageEditorControl) Implements IGraphicCommand.Execute
            editorControl.SetActiveTool(New WatermarkToolControl())
        End Sub
    End Class
    Public Class WatermarkPreset
		Inherits WatermarkCommand

		Public Overrides ReadOnly Property ToolTip() As String
			Get
				Return "Watermark Preset"
			End Get
		End Property
		Public Overrides Sub Execute(ByVal editorControl As ImageEditorControl)
			' Perform the WatermarkGraphicOperation operation in code, using the EditController. Users can undo this operation while the Image Editor is active.
			editorControl.EditController.DoOperation(New WatermarkGraphicOperation("devexpress.com", Color.LightBlue, 15, True))
		End Sub
	End Class
End Namespace
